### PR TITLE
Fetch logins from Connect My Computer role in Discover flow

### PR DESCRIPTION
**Pull request by ravicious**
tl;dr Instead of grabbing all available logins for the node, we grab only the logins from the Connect My Computer role.

## Why

When I originally implemented the temporary connection test step in #34452, I realized that the Web UI doesn't have any way to read the contents of the roles. This shouldn't have been a surprise to me as the frontend in Connect doesn't have such API either. We read details from roles only within lib/teleterm, which is Connect's equivalent of lib/web.

Since I didn't have much time when working on #34452, I just made it so that we fetch the node details again which shows all available logins for this node. But might include logins which we know are not correct, such as root. We could make the UX of it so much better.

## How

Teleport Connect starts the agent as the system user who started Teleport Connect. Teleport Connect also adds that system user as an allowed login to the Connect My Computer role. So if my cluster username is `alice@example.com` and my system username is `alice`, I know that the role `connect-my-computer-alice@example.com` is going to include `alice` in its logins and that it'll be the only valid login for the Connect My Computer node.

So instead of showing all logins, I can just fetch the logins from `connect-my-computer-alice@example.com`.

One edge case is that the role can have multiple logins if the user has set up Connect My Computer on multiple devices as different system users. In this case, we'll let the user choose among the logins in the role.

The actual connection test, which I'm going to work on next (#32206), will use this endpoint to correctly establish the logins for the node as well. It'll be similar to the regular SSH connection test, with the main difference being that it won't ask the user to pick a login if there's only a single login in the Connect My Computer role.

### CODE REVIEW

File: web/packages/shared/constants.ts
Comment (ravicious) [inline web/packages/shared/constants.ts:1]: I removed this file because I felt that it's better to define this stuff in dedicated folders & files, like `shared/connectMyComputer.ts` for example.

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx [hunk 1]
```diff
@@ -94,36 +94,68 @@ export const TestConnection = (props: AgentStepProps) => {
...
-            <TextIcon mt={2} mb={3}>
-              <Icons.Warning size="medium" ml={1} mr={2} color="error.main" />
-              Encountered Error: {refetchNodeAttempt.statusText}
-            </TextIcon>
-
-            <ButtonPrimary
-              type="button"
-              onClick={() => refetchNode(abortController.current.signal)}
+        {fetchLoginsAttempt.status === '' ||
+          (fetchLoginsAttempt.status === 'processing' && <Indicator />)}
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:98]: just double checking that this check was intended or is it:


```suggestion
        {(fetchLoginsAttempt.status === '' ||
          fetchLoginsAttempt.status === 'processing') && <Indicator />}
```

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx [hunk 2]
```diff
@@ -94,36 +94,68 @@ export const TestConnection = (props: AgentStepProps) => {
...
+
+        {fetchLoginsAttempt.status === 'error' &&
+          (fetchLoginsAttempt.error instanceof ApiError &&
+          fetchLoginsAttempt.error.response.status === 404 ? (
+            <ErrorWithinStep
+              buttonText="Reload"
+              buttonOnClick={() => window.location.reload()}
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:105]: i think else where the language we use is `refresh` 🤷‍♀️

File: web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx [hunk 3]
```diff
@@ -271,7 +265,7 @@ const KubeConnect = ({ kube }: { kube: Kube }) => {
   );
 };
 
-const sortNodeLogins = (logins: string[]) => {
+export const sortNodeLogins = (logins: string[]) => {
```
Comment (kimlisa) [inline web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx:268]: perhaps you already had this in mind, should we re-use this: https://github.com/gravitational/teleport/pull/34952/files#diff-b85276b287d12bb5c6ace6e14c78de19476484e30f16dee449546c7d2b7e0d04R55

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx [hunk 4]
```diff
@@ -16,65 +16,65 @@
...
+import cfg from 'teleport/config';
 import useTeleport from 'teleport/useTeleport';
 import {
   ActionButtons,
   StyledBox,
   Header,
   TextIcon,
 } from 'teleport/Discover/Shared';
-import { NodeConnect } from 'teleport/UnifiedResources/ResourceActionButton';
+import { sortNodeLogins } from 'teleport/UnifiedResources/ResourceActionButton';
```
Comment (kimlisa) [inline web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:40]: same here re-use from your other PR?

**Review by kimlisa — APPROVED** [review]

File: constants.go [hunk 5]
```diff
@@ -667,7 +667,7 @@ const (
 
 	// ConnectMyComputerRoleNamePrefix is the prefix used for roles prepared for individual users
 	// during the setup of Connect My Computer. The prefix is followed by the name of the cluster
-	// user. See [teleterm.connectmycomputer.RoleSetup].
+	// user. See teleterm.connectmycomputer.RoleSetup.
```
[the remark this answers is not in the document]
Comment (ravicious) [reply constants.go:670]: That works only if `Features` is in the scope, see https://github.com/gravitational/teleport/pull/28891#discussion_r1261581567

In constants.go, the teleterm package is not imported. Because of this, I'd have to write it out as `[github.com/gravitational/teleport/lib/teleterm/connectmycomputer.RoleSetup]`, which destroys readability IMHO.

File: web/packages/shared/connectMyComputer.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * NodeOwnerLabel is a label used to control access to the node managed by Teleport Connect as part
+ * of Connect My Computer. The agent is configured to have this label and the value of it is the
+ * username of the cluster user that is running the agent.
+ */
+export const NodeOwnerLabel = `teleport.dev/connect-my-computer/owner` as const;
```
[the remark this answers is not in the document]
Comment (ravicious) [reply web/packages/shared/connectMyComputer.ts:22]: I've always thought that strings in JS are mutable just like in Ruby, but they aren't. 😶‍🌫️ I've been used to adding `as const` to arrays and assumed that strings need that too. Good catch, thanks.

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx (hunk 1 above)
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:98]: Oh, it definitely should be what you suggested. The current version will not show anything if status is `''`. I must have not noticed it because the status is equal to `''` just for a split second.

File: web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx (hunk 2 above)
Comment (ravicious) [reply web/packages/teleport/src/Discover/ConnectMyComputer/TestConnection/TestConnection.tsx:105]: Cool, I'll update it to reflect that.

File: web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx (hunk 3 above)
Comment (ravicious) [reply web/packages/teleport/src/UnifiedResources/ResourceActionButton.tsx:268]: Yeah, this PR was made before the other one. In the end I'll make it so that we just share a single function.

**Review by gzdunek — APPROVED** [review]
With Lisa's comments
